scripts/copy_from_upstream/patches/libjade-kyber-meta.patch

```diff
@@ -0,0 +1,120 @@
+diff --git a/src/crypto_kem/kyber/kyber512/META.yml b/src/crypto_kem/kyber/kyber512/META.yml
+index 000ec75..8282075 100644
+--- a/src/crypto_kem/kyber/kyber512/META.yml
++++ b/src/crypto_kem/kyber/kyber512/META.yml
+@@ -1,7 +1,6 @@
+ name: Kyber512
+ type: kem
+-checksumsmall: 9c1a84c0573d21b5fb50ff68f015c19206cebbda4aa3caa6f9ba4b167eea9514
+-checksumbig: 4596232083e3da10d341576afbc59b24a520073e985a9b9df2d587e67e926a7b
++nistkat-sha256: bb0481d3325d828817900b709d23917cefbc10026fc857f098979451f67bb0ca
+ claimed-nist-level: 1
+ claimed-security: IND-CCA2
+ length-public-key: 800
+@@ -9,23 +8,36 @@ length-ciphertext: 768
+ length-secret-key: 1632
+ length-shared-secret: 32
+ principal-submitters:
+-  - TODO
++  - Peter Schwabe
+ auxiliary-submitters:
+-  - TODO
++  - Roberto Avanzi
++  - Joppe Bos
++  - Léo Ducas
++  - Eike Kiltz
++  - Tancrède Lepoint
++  - Vadim Lyubashevsky
++  - John M. Schanck
++  - Gregor Seiler
++  - Damien Stehlé
+ implementations:
+-    - name: amd64/ref
+-      version: TODO
++    - name: ref
++      version: NIST Round 3 submission
```

Comment:
Is it sufficient to use this as version information? Wouldn't it be better to include reference to libjade?

Reply:
This patch is applied to Kyber512/768's `META.yml` files in the libjade repo pulled (and subsequently deleted) by `copy_from_upstream.py`. The information in these YAML files is used by `copy_from_upstream.py` but they don't make their way into liboqs source code in their entirety. 

`docs/algorithms/kem/kyber.yml`, which is populated by `update_upstream_alg_docs.py`, does explicitly list libjade as an upstream and links to the specific release of libjade that liboqs is currently pulling code from (`update_upstream_alg_docs.py` gets this information from `copy_from_libjade.yml` which would need to be updated to pull from newer libjade releases ensuring `docs/algorithms/kem/kyber.yml` is also updated).

`CONFIGURE.md` which documents the `OQS_LIBJADE_BUILD` option (this option must be enabled at build time to include libjade code in liboqs) also links to the libjade repo.

Are there other places you think should also include a reference to libjade?

Reply:
> but they don't make their way into liboqs source code in their entirety.

And this may be a problem: Looking at the resultant .md/.yml files I don't see libjade version information (other than the "Source" reference). Please be aware that whichever "version information finding" logic is required, this must be implemented in [oqs-provider](https://github.com/open-quantum-safe/oqs-provider/blob/main/oqs-template/generate.py) -- or any other possible future downstream caring to make algorithm version information available to users, so having this easily available would be appreciated by anyone (human users and downstream software logic alike) looking at the .yml/.md files.

> Are there other places you think should also include a reference to libjade?

No; to the contrary, there should be as few places as possible to do so, ideally just one -- but that should contain all information easily consumed, something along the lines "Kyber512 - libjade vx.y.z".

Reply:
> Looking at the resultant .md/.yml files I don't see libjade version information (other than the "Source" reference).
>  so having this easily available would be appreciated by anyone (human users and downstream software logic alike) looking at the .yml/.md files.

To my knowledge, this is consistent with how all other upstreams for Kyber are documented at the moment. The "Source" reference includes both a link to the repo and the specific release of libjade the implementation is being pulled from. Do we want to document additional information for libjade implementations?

> "Kyber512 - libjade vx.y.z".

Am I missing where this information is documented for pqclean and pqcrystals implementations of Kyber? Or are do we want to change how libjade implementations are documented? At the moment the .md file lists the upstreams (and their sources) that liboqs pulls from, the algorithm's parameters sets, and the specific implementations liboqs provides for each parameter (each implementation clearly lists its upstream). I'm not sure why we would want to list

Reply:
I've always interpreted these version strings as tracking the version of the algorithm specification that each implementation implements, which in this case is the Round 3 version.